source/frontend/src/features/mapSideBar/acquisition/tabs/expropriation/history/ExpropriationEventHistoryContainer.tsx

```diff
@@ -0,0 +1,184 @@
+  IExpropriationEventHistoryContainerProps
+> = ({ acquisitionFileId, View, ModalView }) => {
+  const [payeeOptions, setPayeeOptions] = useState<PayeeOption[]>([]);
+  const [showExpropriationEditModal, setShowExpropriationEditModal] = useState<boolean>(false);
+  const [editExpropiationEventValue, setEditExpropiationEventValue] = useState(
+    ExpropriationEventFormModel.createEmpty(acquisitionFileId),
+  );
+
+  const { setModalContent, setDisplayModal } = useModalContext();
+
+  const {
+    getAcquisitionOwners: { execute: retrieveAcquisitionOwners, loading: loadingAcquisitionOwners },
+  } = useAcquisitionProvider();
+  const {
+    getAcquisitionInterestHolders: {
+      execute: fetchInterestHolders,
+      loading: loadingInterestHolders,
+    },
+  } = useInterestHolderRepository();
+  const {
+    getExpropriationEvents: {
+      execute: getExpropriationEvents,
+      loading: loadingExpropriationEvents,
+      response: expropriationEventsResponse,
+    },
+    addExpropriationEvent: { execute: addExpropriationEvent },
+    updateExpropriationEvent: { execute: updateExpropriationEvent },
+    deleteExpropriationEvent: { execute: deleteExpropriationEvent },
+  } = useExpropriationEventRepository();
+
+  const expropriationEvents: ApiGen_Concepts_ExpropriationEvent[] =
+    expropriationEventsResponse ?? [];
+
+  const fetchPayeeOptions = useCallback(
+    async (acquisitionFileId: number) => {
+      const acquisitionOwnersCall = retrieveAcquisitionOwners(acquisitionFileId);
+      const interestHoldersCall = fetchInterestHolders(acquisitionFileId);
+
+      await Promise.all([acquisitionOwnersCall, interestHoldersCall]).then(
+        ([acquisitionOwners, interestHolders]) => {
+          const options = [];
+
+          if (acquisitionOwners !== undefined) {
+            const ownersOptions: PayeeOption[] = acquisitionOwners.map(x =>
+              PayeeOption.createOwner(x, null),
+            );
+            options.push(...ownersOptions);
+          }
+
+          if (interestHolders !== undefined) {
+            const interestHolderOptions: PayeeOption[] = interestHolders.map(x =>
+              PayeeOption.createInterestHolder(x, null),
+            );
+            options.push(...interestHolderOptions);
+          }
+
+          setPayeeOptions(options);
+        },
+      );
+    },
+    [retrieveAcquisitionOwners, fetchInterestHolders],
+  );
+
+  // Load the expropriation event history (and payee options) upon rendering this component
+  useEffect(() => {
+    if (isValidId(acquisitionFileId)) {
+      getExpropriationEvents(acquisitionFileId);
+      fetchPayeeOptions(acquisitionFileId);
+    }
+  }, [acquisitionFileId, fetchPayeeOptions, getExpropriationEvents]);
+
+  // Launch the expropriation modal in response to the user clicking the "Add" button
+  const onAddExpropriation = () => {
+    setEditExpropiationEventValue(ExpropriationEventFormModel.createEmpty(acquisitionFileId));
```

Comment:
nit: consider placing both of these into a single state var with multiple fields.